app/src/test/java/ch/hikemate/app/model/elevation/ElevationServiceRepositoryTest.kt

```diff
@@ -0,0 +1,318 @@
+
+    val call2 = mock(Call::class.java)
+    `when`(client.newCall(any())).thenReturn(call2)
+
+    elevationServiceRepository.getElevation(latLong, 0, { list -> assertEquals(longList, list) }) {
+      fail("Failed to fetch routes from Overpass API")
+    }
+    assertEquals(1, elevationServiceRepository.getCacheSize())
+    verify(client).newCall(any())
+  }
+
+  @Test
+  fun cacheWorks_toStoreDifferentIDs() {
+    // Prepare mock responses
+    val call = mock(Call::class.java)
+    `when`(client.newCall(any())).thenReturn(call)
+
+    val callbackCapture = argumentCaptor<okhttp3.Callback>()
+
+    // Simulate the first API response
+    `when`(call.enqueue(callbackCapture.capture())).thenAnswer {
+      // Return the mocked longResponse on the first call
+      callbackCapture.firstValue.onResponse(call, longResponse)
+    }
+
+    // Check initial cache size
+    assertEquals(0, elevationServiceRepository.getCacheSize())
+
+    // First API call
+    elevationServiceRepository.getElevation(latLong, 0, { list -> assertEquals(longList, list) }) {
+      fail("Failed to fetch routes from Overpass API")
+    }
+
+    // Check cache size after first call
+    assertEquals(1, elevationServiceRepository.getCacheSize())
+
+    // Prepare for second call with a new mock
+    val call2 = mock(Call::class.java)
+    `when`(client.newCall(any())).thenReturn(call2)
+
+    val callbackCapture2 = argumentCaptor<okhttp3.Callback>()
+
+    // Simulate the second API response
+    `when`(call2.enqueue(callbackCapture2.capture())).thenAnswer {
+      // Return the same mocked longResponse for the second call
+      callbackCapture2.firstValue.onResponse(call2, simpleResponse)
+    }
+
+    // Second API call
+    elevationServiceRepository.getElevation(latLong, 1, { list -> assertEquals(smallList, list) }) {
+      fail("Failed to fetch routes from Overpass API")
+    }
+
+    // Verify cache size after second call
+    assertEquals(2, elevationServiceRepository.getCacheSize())
+
+    // Verify that newCall was invoked twice
+    verify(client, times(2)).newCall(any())
+  }
+
+  @Test
+  fun failsWithErrorResponse() {
+    // Mock an error response
+    val responseBody = ResponseBody.create("application/json; charset=utf-8".toMediaType(), "")
+    val response =
+        Response.Builder()
+            .code(500)
+            .message("Internal Server Error")
+            .protocol(Protocol.HTTP_1_1)
+            .request(Request.Builder().url("https://api.open-elevation.com/api/v1/lookup").build())
+            .body(responseBody)
+            .build()
+
+    // Mock the call
+    val call = mock(Call::class.java)
+    `when`(call.execute()).thenReturn(response)
+    `when`(client.newCall(any())).thenReturn(call)
+
+    elevationServiceRepository.getElevation(
+        latLong, 0, { fail("Expected failure but got success") }) { exception ->
+          assertEquals("Failed to get elevation", exception.message)
```

Comment:
Nitpick: I don't think the message is relevant here. We just want to make sure the error is called. We don't want to change the test if we later change the exception message.

Reply:
Fixed.